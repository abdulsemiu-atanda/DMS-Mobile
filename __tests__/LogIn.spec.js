import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'

import authScreenProps, {authLoadingProps} from '../util/fixtures/authFixtures'
import {LogIn} from '../components/LogIn.react'

describe('LogIn', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<LogIn {...authScreenProps('login')} />).toJSON()

    expect(tree).toMatchSnapshot()
    expect(tree.children[0].children[0].type).toEqual('Text')
    expect(tree.children[0].children[0].children[0]).toEqual('DMS')
    expect(tree.children[0].children[2].type).toEqual('Text')
    expect(tree.children[0].children[2].children[0]).toEqual('Manage your documents elegantly.')
  })

  it('render correct error message when login fails', () => {
    const failedLoginProps = {
      ...authScreenProps('login'),
      user: {loggingIn: false, message: 'Username or password incorrect'}
    }
    const tree = renderer.create(<LogIn {...failedLoginProps} />).toJSON()

    expect(tree).toMatchSnapshot()
    expect(tree.children[1].children[5].children[0]).toEqual('Username or password incorrect')
  })

  it('renders spinner when user is loggingIn', () => {
    const tree = renderer.create(<LogIn {...authLoadingProps('login')} />).toJSON()

    expect(tree).toMatchSnapshot()
    expect(tree.children[1].children[4].children[0].type).toEqual('ActivityIndicator')
  })
})
