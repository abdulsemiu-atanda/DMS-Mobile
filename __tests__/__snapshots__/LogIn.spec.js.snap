// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogIn render correct error message when login fails 1`] = `
<View
  style={
    Object {
      "backfaceVisibility": "hidden",
      "flex": 1,
      "height": 1334,
      "opacity": 90,
      "transform": Array [
        Object {
          "rotateY": "0deg",
        },
      ],
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#01f0b3",
        "flex": 3,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#f7f7f7",
          "fontSize": 25,
          "fontWeight": "700",
          "textAlign": "center",
        }
      }
    >
      DMS
    </Text>
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#f7f7f7",
            "fontSize": 100,
          },
          Object {
            "textAlign": "center",
          },
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 15,
          "textAlign": "center",
        }
      }
    >
      Manage your documents elegantly.
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 2,
        "padding": 30,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#96a1ae",
          "fontSize": 10,
        }
      }
    >
      USERNAME
    </Text>
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      keyboardType="email-address"
      placeholder="jason@bourne.com"
      style={
        Object {
          "borderBottomColor": "#bbb",
          "borderBottomWidth": 1,
          "marginBottom": 20,
          "marginTop": 10,
        }
      }
      underlineColorAndroid="transparent"
    />
    <Text
      style={
        Object {
          "color": "#96a1ae",
          "fontSize": 10,
        }
      }
    >
      PASSWORD
    </Text>
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      secureTextEntry={true}
      style={
        Object {
          "borderBottomColor": "#bbb",
          "borderBottomWidth": 1,
          "marginBottom": 20,
          "marginTop": 10,
        }
      }
      underlineColorAndroid="transparent"
    />
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#0983ff",
          "height": 40,
          "marginTop": 20,
          "padding": 7,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#aacfff",
            "fontSize": 18,
          }
        }
      >
        Sign In
      </Text>
    </View>
    <Text
      style={
        Object {
          "color": "red",
          "marginTop": 10,
        }
      }
    >
      Username or password incorrect
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
          "padding": 30,
        },
        Object {
          "height": "16%",
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#96a1ae",
          "fontSize": 12,
          "fontWeight": "600",
          "marginRight": 5,
        }
      }
    >
      DON'T HAVE AN ACCOUNT?
    </Text>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 12,
            "fontWeight": "700",
          }
        }
      >
        SIGN UP
      </Text>
    </View>
  </View>
</View>
`;

exports[`LogIn renders correctly 1`] = `
<View
  style={
    Object {
      "backfaceVisibility": "hidden",
      "flex": 1,
      "height": 1334,
      "opacity": 90,
      "transform": Array [
        Object {
          "rotateY": "0deg",
        },
      ],
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#01f0b3",
        "flex": 3,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#f7f7f7",
          "fontSize": 25,
          "fontWeight": "700",
          "textAlign": "center",
        }
      }
    >
      DMS
    </Text>
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#f7f7f7",
            "fontSize": 100,
          },
          Object {
            "textAlign": "center",
          },
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 15,
          "textAlign": "center",
        }
      }
    >
      Manage your documents elegantly.
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 2,
        "padding": 30,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#96a1ae",
          "fontSize": 10,
        }
      }
    >
      USERNAME
    </Text>
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      keyboardType="email-address"
      placeholder="jason@bourne.com"
      style={
        Object {
          "borderBottomColor": "#bbb",
          "borderBottomWidth": 1,
          "marginBottom": 20,
          "marginTop": 10,
        }
      }
      underlineColorAndroid="transparent"
    />
    <Text
      style={
        Object {
          "color": "#96a1ae",
          "fontSize": 10,
        }
      }
    >
      PASSWORD
    </Text>
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      secureTextEntry={true}
      style={
        Object {
          "borderBottomColor": "#bbb",
          "borderBottomWidth": 1,
          "marginBottom": 20,
          "marginTop": 10,
        }
      }
      underlineColorAndroid="transparent"
    />
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#0983ff",
          "height": 40,
          "marginTop": 20,
          "padding": 7,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#aacfff",
            "fontSize": 18,
          }
        }
      >
        Sign In
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
          "padding": 30,
        },
        Object {
          "height": "16%",
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#96a1ae",
          "fontSize": 12,
          "fontWeight": "600",
          "marginRight": 5,
        }
      }
    >
      DON'T HAVE AN ACCOUNT?
    </Text>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 12,
            "fontWeight": "700",
          }
        }
      >
        SIGN UP
      </Text>
    </View>
  </View>
</View>
`;

exports[`LogIn renders spinner when user is loggingIn 1`] = `
<View
  style={
    Object {
      "backfaceVisibility": "hidden",
      "flex": 1,
      "height": 1334,
      "opacity": 90,
      "transform": Array [
        Object {
          "rotateY": "0deg",
        },
      ],
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#01f0b3",
        "flex": 3,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#f7f7f7",
          "fontSize": 25,
          "fontWeight": "700",
          "textAlign": "center",
        }
      }
    >
      DMS
    </Text>
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#f7f7f7",
            "fontSize": 100,
          },
          Object {
            "textAlign": "center",
          },
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 15,
          "textAlign": "center",
        }
      }
    >
      Manage your documents elegantly.
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 2,
        "padding": 30,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#96a1ae",
          "fontSize": 10,
        }
      }
    >
      USERNAME
    </Text>
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      keyboardType="email-address"
      placeholder="jason@bourne.com"
      style={
        Object {
          "borderBottomColor": "#bbb",
          "borderBottomWidth": 1,
          "marginBottom": 20,
          "marginTop": 10,
        }
      }
      underlineColorAndroid="transparent"
    />
    <Text
      style={
        Object {
          "color": "#96a1ae",
          "fontSize": 10,
        }
      }
    >
      PASSWORD
    </Text>
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      secureTextEntry={true}
      style={
        Object {
          "borderBottomColor": "#bbb",
          "borderBottomWidth": 1,
          "marginBottom": 20,
          "marginTop": 10,
        }
      }
      underlineColorAndroid="transparent"
    />
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#0983ff",
          "height": 40,
          "marginTop": 20,
          "padding": 7,
        }
      }
    >
      <ActivityIndicator
        animating={true}
        color="#01f0b3"
        hidesWhenStopped={true}
        size="large"
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
          "padding": 30,
        },
        Object {
          "height": "16%",
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#96a1ae",
          "fontSize": 12,
          "fontWeight": "600",
          "marginRight": 5,
        }
      }
    >
      DON'T HAVE AN ACCOUNT?
    </Text>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 12,
            "fontWeight": "700",
          }
        }
      >
        SIGN UP
      </Text>
    </View>
  </View>
</View>
`;
